docs/default.nix

```diff
@@ -50,35 +50,30 @@ let
     }).optionsJSON;
 
 in
-{
-  inherit options-json;
-  inherit (pkgs) nixos-render-docs;
+lib.fix (
+  self:
+  {
+    inherit options-json;
+    inherit (pkgs) nixos-render-docs;
 
-  gfm-alerts-to-admonitions = pkgs.python3.pkgs.callPackage ./gfm-alerts-to-admonitions { };
+    gfm-alerts-to-admonitions = pkgs.python3.pkgs.callPackage ./gfm-alerts-to-admonitions { };
 
-  man-docs = pkgs.callPackage ./man { inherit options-json; };
-}
-// lib.optionalAttrs (!pkgs.stdenv.isDarwin) (
-  let
-    mkSearch =
-      baseHref:
-      nuschtosSearch.packages.mkSearch {
-        optionsJSON = options-json + "/share/doc/nixos/options.json";
-        urlPrefix = "https://github.com/nix-community/nixvim/tree/main";
-        title = "Nixvim options search";
-        inherit baseHref;
-      };
-  in
-  {
+    man-docs = pkgs.callPackage ./man { inherit options-json; };
+  }
+  // lib.optionalAttrs (!pkgs.stdenv.isDarwin) {
     # NuschtOS/search does not seem to work on darwin
-    search = mkSearch "/";
+    search = nuschtosSearch.packages.mkSearch {
+      optionsJSON = options-json + "/share/doc/nixos/options.json";
+      urlPrefix = "https://github.com/nix-community/nixvim/tree/main";
```

Comment:
We should maybe the github URL overrideable too?

Reply:
I'm more thinking of end-users who may wish to build the docs locally themselves. 

Or if we want to locally build them to test out a backport or whatever.

But sure, it's not too important to get perfect straight away